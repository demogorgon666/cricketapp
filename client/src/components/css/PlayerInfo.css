@import url("https://fonts.googleapis.com/css?family=Montserrat:400,500,600&display=swap");

.mainDiv {
  display: grid;
  grid-template-columns: 50% 50%;
  margin-top: 200px;
}

.personalDetails {
  margin-top: 1.5em;
  border-right: 1px solid #ccc;
  /* height: 500px; */
  justify-items: flex-start;
  /* border: 1px solid #ccc; */
  height: 500px;
}

.playerBasicinfo {
  display: flex;
  flex-direction: column;
  text-align: left;
  margin: 8px;
}

.careerStats {
  margin-top: 1.5em;
  justify-items: flex-start;
  height: 500px;
}

.formatHeading {
  /* height: 40px; */
  background-color: #2980b9;
}
