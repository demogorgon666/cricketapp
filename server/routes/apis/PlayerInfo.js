const express = require("express");
const router = express.Router();
const pg = require("pg-promise")();
// const config = require("config");

// const postgresURL = config.get("postgresURL");
const db = pg("postgres://postgres:root@localhost:5432/cricketalpha");

router.get("/allPlayer", async (req, res) => {
  try {
    const result = await db.any("select * from player");
    res.status(200).json({
      status: 200,
      data: result,
      message: "Retrieved all players"
    });
  } catch (err) {
    console.log(err);
  }
});

router.get("/singlePlayer/:player_id", async (req, res) => {
  const id = req.params.player_id;
  console.log("player id is: ", id);
  try {
    const result = await db.any(
      `select * from player where player_id = ${id} `
    );

    if (result.length < 1)
      throw {
        statusCode: res.status(400).json({
          statusCode: 400,
          statusMessage: "ERROR!Bad Request cannot retrieve all players"
        })
      };

    res.status(200).json({
      status: 200,
      data: result,
      message: "player retrieved succefully"
    });
  } catch (err) {
    console.log(err);
  }
});

router.get("/allBatsman", async (req, res) => {
  try {
    var BatRole1 = " Batsman ";
    var BatRole2 = " WK-Batsman ";
    var BatRole3 = " Batting Allrounder ";
    const result = await db.any(
      "select * from player where player_role ='" +
        BatRole1 +
        "'OR player_role = '" +
        BatRole2 +
        "'OR player_role = '" +
        BatRole3 +
        "' ;"
    );
    // console.log("data is: ", result);
    res.status(200).json({
      status: 200,
      data: result,
      message: "All Batsman retrieved"
    });
  } catch (err) {
    console.log(err);
  }
});

router.get("/allBowlers", async (req, res) => {
  try {
    var BallRole1 = " Bowler ";
    var BallRole2 = " Bowling Allrounder ";
    const result = await db.any(
      "select * from player where player_role ='" +
        BallRole1 +
        "'OR player_role = '" +
        BallRole2 +
        "' ;"
    );
    // console.log("data is: ", result);
    res.status(200).json({
      status: 200,
      data: result,
      message: "All Bowlers retrieved"
    });
  } catch (err) {
    console.log(err);
  }
});

router.post("/TopBatsman", async (req, res) => {
  try {
    let match_type = req.body.match_type;
    console.log(match_type);
    const result = await db.any(
      "select player_stats.match_type,player_stats.player_stats_name, player_stats.player_stats_value,player.player_name,player.player_id from player_stats inner join player on player_stats.player_id = player.player_id where player_stats.match_type = '" +
        match_type +
        "'AND player_stats_name = 'total_runs' order by player_stats_value desc fetch first 5 rows only"
    );

    // console.log("data is: ", result);
    res.status(200).json({
      status: 200,
      data: result,
      message: "All top Batsman retrieved"
    });
  } catch (err) {
    console.log(err);
  }
});

router.post("/TopBowlers", async (req, res) => {
  try {
    let match_type = req.body.match_type;

    const result = await db.any(
      "select player_stats.match_type,player_stats.player_stats_name, player_stats.player_stats_value,player.player_name from player_stats inner join player on player_stats.player_id = player.player_id where player_stats.match_type = '" +
        match_type +
        "'AND player_stats_name = 'total_wickets' order by player_stats_value desc fetch first 5 rows only"
    );

    console.log("data is: ", result);
    res.status(200).json({
      status: 200,
      data: result,
      message: "All top bowlers retrieved"
    });
  } catch (err) {
    console.log(err);
  }
});

module.exports = router;
