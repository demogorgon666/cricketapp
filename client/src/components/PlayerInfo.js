import React, { Component } from "react";
import "./css/PlayerInfo.css";
import Navbar from "./common/Navbar";
import "../components/css/SecondaryNavbar.css";
import dhoni from "../components/images/dhoni.jpg";
import { Link } from "react-router-dom";
import PlayerBattingStats from "./PlayerBattingStats";
import PlayerBowlingStats from "./PlayerBowlingStats";

import { getSinglePlayer } from "../actions/PlayerAction";
import { connect } from "react-redux";

export class PlayerInfo extends Component {
  state = {
    player_id: "",
    info: true,
    batting_stats: false,
    bowling_stats: false
  };

  componentDidMount() {
    this.props.getSinglePlayer(this.props.match.params.player_id);
  }

  render() {
    console.log(this.props);
    console.log(this.props.singlePlayer[0].player_image);
    // console.log(
    //   "Inside render component",
    //   this.props.singlePlayer[0].player_dob
    // );
    if (this.state.info === true)
      return (
        <div>
          <Navbar
            gender={this.props.gender}
            changeGender={getGender => this.props.changeGender(getGender)}
          />

          <div className="secNavParent">
            <div className="secNavHead">
              <div
                style={{
                  backgroundImage: `url(data:image/jpeg;base64,${this.props.singlePlayer[0].player_image})`
                }}
              ></div>
              <div>
                <h2>{this.props.singlePlayer[0].player_name}</h2>
                <h4 style={{ textAlign: "left" }}>
                  {this.props.singlePlayer[0].player_country}
                </h4>
              </div>
            </div>
            <div className="secNavLinks">
              <ul>
                <Link
                  className="secLink"
                  onClick={() => {
                    this.setState({
                      info: true,
                      batting_stats: false,
                      bowling_stats: false
                    });
                  }}
                >
                  <li>Info</li>
                </Link>
                <Link
                  className="secLink"
                  onClick={() => {
                    this.setState({
                      info: false,
                      batting_stats: true,
                      bowling_stats: false
                    });
                  }}
                >
                  <li
                    className={
                      this.state.battingSelected ? "sec-active" : "not-active"
                    }
                  >
                    Batting Stats
                  </li>
                </Link>
                <Link
                  className="secLink"
                  onClick={() => {
                    this.setState({
                      info: false,
                      batting_stats: false,
                      bowling_stats: true
                    });
                  }}
                >
                  <li>Bowling Stats</li>
                </Link>
              </ul>
            </div>
          </div>
          <div className="mainDiv" style={{ padding: 16 + "px" }}>
            <div
              className="personalDetails"
              style={{ fontFamily: "Montserrat", color: "#272727" }}
            >
              <p
                style={{
                  borderBottom: "1.2px solid",
                  borderBottomColor: "#ccc",
                  width: 200 + "px",
                  fontSize: 20 + "px",
                  textAlign: "left"
                }}
              >
                Personal Details
              </p>
              <div className="playerBasicinfo">
                <span id="jest-full-name" style={{ fontSize: 14 + "px" }}>
                  Full Name
                </span>
                <b>{this.props.singlePlayer[0].player_name}</b>
                <div style={{ marginTop: 16 + "px" }}>
                  <span>Date Of Birth</span>
                  {/* <span
                  style={{
                    fontSize: 14 + "px",
                    float: "right",
                    marginRight: 380
                  }}
                >
                  Age
                </span> 
                {this.props.singlePlayer[0].player_dob}
                */}
                </div>
                <b>{this.props.singlePlayer[0].player_dob}</b>
                <span id="jest-role">Role</span>
                <b>{this.props.singlePlayer[0].player_role}</b>
                <span id="jest-batting-style">Batting Style</span>
                <b>{this.props.singlePlayer[0].batting_style}</b>
                <span id="jest-bowling-style">Bowling Style</span>
                <b>{this.props.singlePlayer[0].bowling_style}</b>
              </div>
            </div>
            <div className="careerStats" style={{ fontFamily: "Montserrat" }}>
              <p
                id="career-stats-heading"
                style={{
                  borderBottom: "1.2px solid",
                  borderBottomColor: "#ccc",
                  width: 200 + "px",
                  fontSize: 20 + "px",
                  textAlign: "left"
                }}
              >
                Career Statistics
              </p>
              <div>
                <div className="formatHeading">
                  <span id="test-heading">Test</span>
                </div>
                <div
                  className="pl-stats-details"
                  style={{
                    textAlign: "left",
                    // marginTop: 15 + "px",
                    // marginLeft: 15 + "px",
                    // marginBottom: 15 + "px",
                    padding: 6 + "px",
                    backgroundColor: "white"
                  }}
                >
                  <p
                    style={{
                      margin: 6 + "px",
                      fontSize: 14 + "px",
                      fontWeight: 600,
                      color: "#707070"
                    }}
                  >
                    Debut
                  </p>
                  <b>{this.props.singlePlayer[0].debut_test_match}</b>
                </div>

                <div className="formatHeading">
                  <span id="odi-heading">ODI</span>
                </div>
                <div
                  className="pl-stats-details"
                  style={{
                    textAlign: "left",
                    padding: 6 + "px",
                    backgroundColor: "white"
                  }}
                >
                  <p
                    style={{
                      margin: 6 + "px",
                      fontSize: 14 + "px",
                      fontWeight: 600,
                      color: "#707070"
                    }}
                  >
                    Debut
                  </p>
                  <b>{this.props.singlePlayer[0].debut_odi_match}</b>
                </div>

                <div className="formatHeading">
                  <span id="t20-heading">T20</span>
                </div>
                <div
                  className="pl-stats-details"
                  style={{
                    textAlign: "left",

                    backgroundColor: "white"
                  }}
                >
                  <p
                    style={{
                      margin: 6 + "px",
                      fontSize: 14 + "px",
                      fontWeight: 600,
                      color: "#707070"
                    }}
                  >
                    Debut
                  </p>
                  <b>{this.props.singlePlayer[0].debut_t20_match}</b>
                </div>
              </div>
            </div>
          </div>
        </div>
      );
    else if (this.state.batting_stats === true) {
      return (
        <div>
          <Navbar
            gender={this.props.gender}
            changeGender={getGender => this.props.changeGender(getGender)}
          />

          <div className="secNavParent">
            <div className="secNavHead">
              <div
                style={{
                  backgroundImage: `url(data:image/jpeg;base64,${this.props.singlePlayer[0].player_image})`
                }}
              ></div>
              <div>
                <h2>{this.props.singlePlayer[0].player_name}</h2>
                <h4 style={{ textAlign: "left" }}>
                  {this.props.singlePlayer[0].player_country}
                </h4>
              </div>
            </div>
            <div className="secNavLinks">
              <ul>
                <Link
                  className="secLink"
                  onClick={() => {
                    this.setState({
                      info: true,
                      batting_stats: false,
                      bowling_stats: false
                    });
                  }}
                >
                  <li>Info</li>
                </Link>
                <Link
                  className="secLink"
                  onClick={() => {
                    this.setState({
                      info: false,
                      batting_stats: true,
                      bowling_stats: false
                    });
                  }}
                >
                  <li>Batting Stats</li>
                </Link>
                <Link
                  className="secLink"
                  onClick={() => {
                    this.setState({
                      info: false,
                      batting_stats: false,
                      bowling_stats: true
                    });
                  }}
                >
                  <li>Bowling Stats</li>
                </Link>
              </ul>
            </div>
          </div>
          <PlayerBattingStats
            batsmanCareerStats={this.props.batsmanStats}
            player_id={this.props.match.params.player_id}
          />
        </div>
      );
    } else if (this.state.bowling_stats === true) {
      return (
        <div>
          <Navbar
            gender={this.props.gender}
            changeGender={getGender => this.props.changeGender(getGender)}
          />

          <div className="secNavParent">
            <div className="secNavHead">
              <div
                style={{
                  backgroundImage: `url(data:image/jpeg;base64,${this.props.singlePlayer[0].player_image})`
                }}
              ></div>
              <div>
                <h2>{this.props.singlePlayer[0].player_name}</h2>
                <h4 style={{ textAlign: "left" }}>
                  {this.props.singlePlayer[0].player_country}
                </h4>
              </div>
            </div>
            <div className="secNavLinks">
              <ul>
                <Link
                  className="secLink"
                  onClick={() => {
                    this.setState({
                      info: true,
                      batting_stats: false,
                      bowling_stats: false
                    });
                  }}
                >
                  <li>Info</li>
                </Link>
                <Link
                  className="secLink"
                  onClick={() => {
                    this.setState({
                      info: false,
                      batting_stats: true,
                      bowling_stats: false
                    });
                  }}
                >
                  <li>Batting Stats</li>
                </Link>
                <Link
                  className="secLink"
                  onClick={() => {
                    this.setState({
                      info: false,
                      batting_stats: false,
                      bowling_stats: true
                    });
                  }}
                >
                  <li>Bowling Stats</li>
                </Link>
              </ul>
            </div>
          </div>
          <PlayerBowlingStats player_id={this.props.match.params.player_id} />
        </div>
      );
    }
  }
}

const mapStateToProps = state => ({
  singlePlayer: state.PlayerReducer.singlePlayer
});

export default connect(
  mapStateToProps,
  { getSinglePlayer }
)(PlayerInfo);
