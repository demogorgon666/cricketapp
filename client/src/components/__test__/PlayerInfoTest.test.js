import React from "react";
import { PlayerInfo } from "../PlayerInfo";
import { shallow } from "enzyme";

const getSinglePlayer = jest.fn();
var singlePlayer = [[]];

const wrapper = shallow(
  <PlayerInfo
    getSinglePlayer={getSinglePlayer}
    singlePlayer={singlePlayer}
    match={{ isExact: true, params: { path: "/", url: "/" } }}
  />
);

describe("Testing PlayerInfo", () => {
  it("should have 18 div", () => {
    expect(wrapper.find("div").length).toBe(18);
  });

  it("should have 8 span elements", () => {
    expect(wrapper.find("span").length).toBe(8);
  });

  it("span elements must have Full Name,Date Of Birth,Test,ODI,T20 text", () => {
    expect(
      wrapper
        .find("span")
        .at(0)
        .text()
    ).toBe("Full Name");
    expect(
      wrapper
        .find("span")
        .at(1)
        .text()
    ).toBe("Date Of Birth");
    expect(
      wrapper
        .find("span")
        .at(2)
        .text()
    ).toBe("Role");
    expect(
      wrapper
        .find("span")
        .at(3)
        .text()
    ).toBe("Batting Style");
    expect(
      wrapper
        .find("span")
        .at(4)
        .text()
    ).toBe("Bowling Style");
    expect(
      wrapper
        .find("span")
        .at(5)
        .text()
    ).toBe("Test");
    expect(
      wrapper
        .find("span")
        .at(6)
        .text()
    ).toBe("ODI");
    expect(
      wrapper
        .find("span")
        .at(7)
        .text()
    ).toBe("T20");
  });

  it("should have 5 p tag elements", () => {
    expect(wrapper.find("p").length).toBe(5);
  });

  it("p tag elements should have Personal Details,Career Statistics,debut,debut,debut text", () => {
    expect(
      wrapper
        .find("p")
        .at(0)
        .text()
    ).toBe("Personal Details");

    expect(
      wrapper
        .find("p")
        .at(1)
        .text()
    ).toBe("Career Statistics");

    expect(
      wrapper
        .find("p")
        .at(2)
        .text()
    ).toBe("Debut");

    expect(
      wrapper
        .find("p")
        .at(3)
        .text()
    ).toBe("Debut");
    expect(
      wrapper
        .find("p")
        .at(4)
        .text()
    ).toBe("Debut");
  });

  it("should have 8 b elements ", () => {
    expect(wrapper.find("b").length).toBe(8);
  });
});
