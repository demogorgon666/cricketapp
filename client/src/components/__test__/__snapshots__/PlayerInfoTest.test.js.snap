// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing PlayerInfo should render the component 1`] = `
<div>
<<<<<<< HEAD
  <Connect(Navbar)
    changeGender={[Function]}
  />
=======
>>>>>>> d39022411bcba92990b0f26073b69fa0c23b8944
  <div
    className="secNavParent"
  >
    <div
      className="secNavHead"
    >
      <div
        style={
          Object {
            "backgroundImage": "url(data:image/jpeg;base64,undefined)",
          }
        }
      />
      <div>
        <h2 />
        <h4
          style={
            Object {
              "textAlign": "left",
            }
          }
        />
      </div>
    </div>
    <div
      className="secNavLinks"
    >
      <ul>
        <Link
          className="secLink"
          onClick={[Function]}
        >
          <li>
            Info
          </li>
        </Link>
        <Link
          className="secLink"
          onClick={[Function]}
        >
          <li
            className="not-active"
          >
            Batting Stats
          </li>
        </Link>
        <Link
          className="secLink"
          onClick={[Function]}
        >
          <li>
            Bowling Stats
          </li>
        </Link>
        <Link
          className="secLink"
          to="/performance"
        >
          <li>
            Performance
          </li>
        </Link>
      </ul>
    </div>
  </div>
  <div
    className="mainDiv"
    style={
      Object {
        "padding": "16px",
      }
    }
  >
    <div
      className="personalDetails"
      style={
        Object {
          "color": "#272727",
          "fontFamily": "Montserrat",
        }
      }
    >
      <p
        style={
          Object {
            "borderBottom": "1.2px solid",
            "borderBottomColor": "#ccc",
            "fontSize": "20px",
            "textAlign": "left",
            "width": "200px",
          }
        }
      >
        Personal Details
      </p>
      <div
        className="playerBasicinfo"
      >
        <span
          id="jest-full-name"
          style={
            Object {
              "fontSize": "14px",
            }
          }
        >
          Full Name
        </span>
        <b />
        <div
          style={
            Object {
              "marginTop": "16px",
            }
          }
        >
          <span>
            Date Of Birth
          </span>
        </div>
        <b />
        <span
          id="jest-role"
        >
          Role
        </span>
        <b />
        <span
          id="jest-batting-style"
        >
          Batting Style
        </span>
        <b />
        <span
          id="jest-bowling-style"
        >
          Bowling Style
        </span>
        <b />
      </div>
    </div>
    <div
      className="careerStats"
      style={
        Object {
          "fontFamily": "Montserrat",
        }
      }
    >
      <p
        id="career-stats-heading"
        style={
          Object {
            "borderBottom": "1.2px solid",
            "borderBottomColor": "#ccc",
            "fontSize": "20px",
            "textAlign": "left",
            "width": "200px",
          }
        }
      >
        Career Statistics
      </p>
      <div>
        <div
          className="formatHeading"
        >
          <span
            id="test-heading"
          >
            Test
          </span>
        </div>
        <div
          className="pl-stats-details"
          style={
            Object {
              "marginBottom": "15px",
              "marginLeft": "15px",
              "marginTop": "15px",
              "textAlign": "left",
            }
          }
        >
          <p
            style={
              Object {
                "color": "#707070",
                "fontSize": "14px",
                "fontWeight": 600,
                "margin": "6px",
              }
            }
          >
            Debut
          </p>
          <b />
        </div>
        <div
          className="formatHeading"
        >
          <span
            id="odi-heading"
          >
            ODI
          </span>
        </div>
        <div
          className="pl-stats-details"
          style={
            Object {
              "color": "#707070",
              "marginBottom": "20px",
              "marginLeft": "20px",
              "marginTop": "20px",
              "textAlign": "left",
            }
          }
        >
          <p
            style={
              Object {
                "color": "#707070",
                "fontSize": "14px",
                "fontWeight": 600,
                "margin": "6px",
              }
            }
          >
            Debut
          </p>
          <b />
        </div>
        <div
          className="formatHeading"
        >
          <span
            id="t20-heading"
          >
            T20
          </span>
        </div>
        <div
          className="pl-stats-details"
          style={
            Object {
              "marginBottom": "20px",
              "marginLeft": "20px",
              "marginTop": "20px",
              "textAlign": "left",
            }
          }
        >
          <p
            style={
              Object {
                "color": "#707070",
                "fontSize": "14px",
                "fontWeight": 600,
                "margin": "6px",
              }
            }
          >
            Debut
          </p>
          <b />
        </div>
      </div>
    </div>
  </div>
</div>
`;
